python/cudf/cudf/core/column_accessor.py

```diff
@@ -32,13 +36,67 @@ def __getitem__(self, key):
 
     def __setitem__(self, key, value):
         self.set_by_label(key, value)
+        self._clear_cache()
```

Comment:
Future optimization: I think it should be possible to just add the item to the `NestedOrderedDict` if there is one instead of clearing its cache here.